portfolio/src/main/webapp/script.js

```diff
@@ -35,7 +35,40 @@ function createListElement(text) {
 }
 
 function createMap() {
-  const map = new google.maps.Map(
-      document.getElementById('map'),
-      {center: {lat: 37.422, lng: -122.084}, zoom: 16});
+  var locations = [
+    ["Beijing", 39.9042, 116.4074],
+    ["Giza", 30.0131, 31.2089],
+    ["Cape Town", -33.9249, 18.4241],
+    ["Tokyo", 35.6762, 139.6503],
+    ["Rio de Janeiro", 36.3932, 25.4615],
+    ["Santorini", 36.3932, 25.4615],
+    ["San Francisco", 37.7749, -122.4194],
+    ["Dubai", 25.2048, 55.2708],
+    ["Paris", 48.8566, 2.3522],
+    ["Rome", 41.9028, 12.4964],
+    ["London", 51.5074, -0.1278],
+    ["Bora Bora", -16.5004, -151.7415],
+    ["Madrid", 40.4168, -3.7038]
+  ];
+
+  var map = new google.maps.Map(
+      document.getElementById("map"),
+      {center: new google.maps.LatLng(0, 0), zoom: 1});
+
+  var infowindow = new google.maps.InfoWindow();
+
+  var marker, i;
+
+  for (i = 0; i < locations.length; i++) {  
+      marker = new google.maps.Marker({
+        position: new google.maps.LatLng(locations[i][1], locations[i][2]),
+        map: map
+      });
+      google.maps.event.addListener(marker, "click", (function(marker, i) {
+        return function() {
+          infowindow.setContent(locations[i][0]);
+          infowindow.open(map, marker);
+        }
+      })(marker, i));
```

Comment:
Nit: you can replace this by a for-in loop -- "for (const location in locations)".

Once you change location to const instead of using "var i" and define the marker inside the loop -- "const marker = new google.maps.Marker ...", you should be able to remove the function closure and simply do:

```suggestion
      google.maps.event.addListener(marker, "click", function() {
        infowindow.setContent(location[0]);
        infowindow.open(map, marker);
      });
```

Reply:
![image](https://user-images.githubusercontent.com/55557781/86316244-4873fb00-bbfa-11ea-8043-6554c133b17f.png)
I tried to do that, but the location wouldn't copy the data over correctly, for example the location.length was 1 for the first 9 of the locations when it should be 3

Reply:
Ah, interesting. Looks like we should use [for of](https://www.w3schools.com/js/js_loop_for.asp#:~:text=The%20For%2FOf%20Loop&text=for%2Fof%20lets%20you%20loop,Maps%2C%20NodeLists%2C%20and%20more.&text=variable%20%2D%20For%20every%20iteration%20the,const%20%2C%20let%20%2C%20or%20var%20.) instead of "for in" for iterating over iterables. I have approved the PR, so feel free to merge the PR if "for of" does not work either.